rotkehlchen/chain/evm/decoding/ens/decoder.py

```diff
@@ -0,0 +1,311 @@
+import logging
+from abc import abstractmethod
+from typing import TYPE_CHECKING
+
+import content_hash
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.utils import TokenEncounterInfo, get_or_create_evm_token
+from rotkehlchen.chain.ethereum.abi import decode_event_data_abi_str
+from rotkehlchen.chain.evm.decoding.constants import ERC20_OR_ERC721_TRANSFER
+from rotkehlchen.chain.evm.decoding.ens.constants import (
+    ADDR_CHANGED,
+    CONTENT_HASH_CHANGED,
+    NEW_OWNER,
+    NEW_RESOLVER,
+    TEXT_CHANGED_KEY_AND_VALUE,
+    TEXT_CHANGED_KEY_AND_VALUE_ABI,
+    TEXT_CHANGED_KEY_ONLY,
+    TEXT_CHANGED_KEY_ONLY_ABI,
+)
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress, EvmTokenKind
+from rotkehlchen.utils.misc import bytes_to_address
+from rotkehlchen.utils.mixins.customizable_date import CustomizableDateMixin
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class EnsCommonDecoder(DecoderInterface, CustomizableDateMixin):
```

Comment:
you did not make it ann abstract class. It needs to inherit from ABC. Check the rest of the code. Check what an abstract class is and don't resolve comments in the PR without doing them or if you don't understand them then ask.

Reply:
Now inherits ABC, but see question in [discord](https://discord.com/channels/657906918408585217/1080783409640648765/1306333798644060281).

Reply:
it is required as explained in discord.